<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>766</width>
    <height>575</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QComboBox" name="comboBox">
    <property name="geometry">
     <rect>
      <x>46</x>
      <y>9</y>
      <width>241</width>
      <height>35</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>Modbus serial master mode</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Modbus serial slave mode</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Modbus TCP master mode</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>Modbus TCP slave mode</string>
     </property>
    </item>
   </widget>
   <widget class="QLineEdit" name="IPlineEdit">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>80</y>
      <width>146</width>
      <height>35</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label">
    <property name="enabled">
     <bool>true</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>78</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>IP address</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_2">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>130</y>
      <width>33</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Port</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="PORTlineEdit">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>160</y>
      <width>91</width>
      <height>35</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="FCspinBox">
    <property name="geometry">
     <rect>
      <x>130</x>
      <y>295</y>
      <width>51</width>
      <height>35</height>
     </rect>
    </property>
    <property name="minimum">
     <number>1</number>
    </property>
    <property name="maximum">
     <number>6</number>
    </property>
   </widget>
   <widget class="QLabel" name="label_3">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>300</y>
      <width>107</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Function code</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="UIDlineEdit">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>240</y>
      <width>91</width>
      <height>35</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_4">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>210</y>
      <width>51</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Unit ID</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_5">
    <property name="geometry">
     <rect>
      <x>185</x>
      <y>130</y>
      <width>59</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Slave ID</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_6">
    <property name="geometry">
     <rect>
      <x>185</x>
      <y>50</y>
      <width>124</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Serial port name</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="COMPORTlineEdit">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>80</y>
      <width>161</width>
      <height>35</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_7">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>355</y>
      <width>176</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Total number requested</string>
    </property>
   </widget>
   <widget class="QLabel" name="label_8">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>138</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>The status to write</string>
    </property>
   </widget>
   <widget class="QLineEdit" name="SLVIDlineEdit">
    <property name="geometry">
     <rect>
      <x>180</x>
      <y>160</y>
      <width>81</width>
      <height>35</height>
     </rect>
    </property>
   </widget>
   <widget class="QLineEdit" name="DATALENlineEdit">
    <property name="geometry">
     <rect>
      <x>200</x>
      <y>350</y>
      <width>101</width>
      <height>35</height>
     </rect>
    </property>
   </widget>
   <widget class="QLabel" name="label_9">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>460</y>
      <width>133</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>The value to write</string>
    </property>
   </widget>
   <widget class="QWidget" name="widget" native="true">
    <property name="geometry">
     <rect>
      <x>350</x>
      <y>10</y>
      <width>411</width>
      <height>491</height>
     </rect>
    </property>
    <widget class="QLabel" name="label_10">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>101</width>
       <height>21</height>
      </rect>
     </property>
     <property name="text">
      <string>Query send :</string>
     </property>
    </widget>
    <widget class="QLabel" name="label_11">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>245</y>
       <width>121</width>
       <height>25</height>
      </rect>
     </property>
     <property name="text">
      <string>Response get :</string>
     </property>
    </widget>
    <widget class="QListWidget" name="QryListWidget">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>40</y>
       <width>381</width>
       <height>200</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
    </widget>
    <widget class="QListWidget" name="RespListWidget">
     <property name="geometry">
      <rect>
       <x>15</x>
       <y>280</y>
       <width>381</width>
       <height>200</height>
      </rect>
     </property>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
     <item>
      <property name="text">
       <string/>
      </property>
     </item>
    </widget>
   </widget>
   <widget class="QLineEdit" name="VALUElineEdit">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>455</y>
      <width>101</width>
      <height>35</height>
     </rect>
    </property>
   </widget>
   <widget class="QComboBox" name="STATUScomboBox">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>405</y>
      <width>61</width>
      <height>35</height>
     </rect>
    </property>
    <item>
     <property name="text">
      <string>ON</string>
     </property>
    </item>
    <item>
     <property name="text">
      <string>OFF</string>
     </property>
    </item>
   </widget>
   <widget class="QDialogButtonBox" name="buttonBox">
    <property name="geometry">
     <rect>
      <x>160</x>
      <y>240</y>
      <width>176</width>
      <height>35</height>
     </rect>
    </property>
    <property name="layoutDirection">
     <enum>Qt::RightToLeft</enum>
    </property>
    <property name="standardButtons">
     <set>QDialogButtonBox::Close|QDialogButtonBox::Open</set>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>766</width>
     <height>33</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
